### PR TITLE
docs: update do-not-disrupt description

### PR DESCRIPTION
**Pull request by jmdeal**
Fixes #N/A 

**Description**
Updates the description for `karpenter.sh/do-not-disrupt` to reflect the changes made when TGP was introduced.

**How was this change tested?**

**Does this change impact docs?**
- [x] Yes, PR includes docs updates 
- [ ] Yes, issue opened: # 
- [ ] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: website/content/en/docs/upgrading/v1-migration.md [hunk 1]
```diff
@@ -411,6 +411,7 @@ Karpenter should now be pulling and operating against the v1beta1 APIVersion as
 * API Rename: NodePool’s ConsolidationPolicy `WhenUnderutilized` is now renamed to `WhenEmptyOrUnderutilized`
 * Behavior Changes:
   * Expiration is now forceful and begins draining as soon as it’s expired. Karpenter does not wait for replacement capacity to be available before draining, but will start provisioning a replacement as soon as the node is expired and begins draining.
+  * Pods with the `karpenter.sh/do-not-disrupt` annotation now block node termination. Termination of a node with these pods will be blocked until those pods are removed, enter a terminating or terminal state, or the NodeClaims's TerminationGracePeriod has expired.
```
Comment (jonathan-innis) [inline website/content/en/docs/upgrading/v1-migration.md:414]: Can we link out to the section of the docs that describes TerminationGracePeriod here?

File: website/content/en/docs/concepts/disruption.md [hunk 2]
```diff
@@ -274,8 +282,14 @@ Duration and Schedule must be defined together. When omitted, the budget is alwa
 
 ### Pod-Level Controls
 
-You can block Karpenter from voluntarily choosing to disrupt certain pods by setting the `karpenter.sh/do-not-disrupt: "true"` annotation on the pod. This is useful for pods that you want to run from start to finish without disruption. By opting pods out of this disruption, you are telling Karpenter that it should not voluntarily remove a node containing this pod.
+You can block Karpenter from voluntarily choosing to disrupt certain pods by setting the `karpenter.sh/do-not-disrupt: "true"` annotation on the pod.
+You can treat this annotation as a single-node, permanently blocking PDB.
```
Comment (jonathan-innis) [inline website/content/en/docs/concepts/disruption.md:286]:
```suggestion
You can treat this annotation as a single-pod, permanently blocking PDB.
```

File: website/content/en/docs/concepts/disruption.md [hunk 3]
```diff
@@ -274,8 +282,14 @@ Duration and Schedule must be defined together. When omitted, the budget is alwa
...
+This has the following consequences:
+- Nodes with `do-not-disrupt` pods will be excluded from **voluntary** disruption, i.e. [Consolidation]({{<ref "#consolidation" >}}) and [Drift]({{<ref "#drift" >}}).
```
Comment (jonathan-innis) [inline website/content/en/docs/concepts/disruption.md:288]: Doesn't voluntary disruption include drift? I'm not sure that they are considered different things here

File: website/content/en/docs/concepts/disruption.md [hunk 4]
```diff
@@ -274,8 +282,14 @@ Duration and Schedule must be defined together. When omitted, the budget is alwa
...
+- Like pods with a blocking PDB, pods with the `do-not-disrupt` annotation will **not** be gracefully evicted by the [Termination Controller]({{ref "#terminationcontroller"}}).
```
Comment (jonathan-innis) [inline website/content/en/docs/concepts/disruption.md:289]:
```suggestion
- Like pods with a fully blocking PDB, pods with the `do-not-disrupt` annotation will **not** be gracefully evicted by the [Termination Controller]({{ref "#terminationcontroller"}}).
```

Comment (jonathan-innis) [inline website/content/en/docs/concepts/disruption.md:289]:
```suggestion
- Like pods with a blocking PDB, pods with the `do-not-disrupt` annotation will **not** be gracefully evicted by the [Termination Controller]({{ref "#terminationcontroller"}}). These pods will either run to completion or be forcefully terminated when the node is near its terminationGracePeriod
```

Consider linking to terminationGracePeriod if you update the docs wording in this way too

File: website/content/en/docs/concepts/disruption.md [hunk 5]
```diff
@@ -274,8 +282,14 @@ Duration and Schedule must be defined together. When omitted, the budget is alwa
...
+  Karpenter will not be able to complete termination of the node until one of the following conditions is met: the `do-not-disrupt` pod is removed, it enters a [terminating](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-phase) or [terminal](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-phase) state, or the owning NodeClaim's [TerminationGracePeriod]({{<ref "#terminationgraceperiod" >}}) has been met.
```
Comment (jonathan-innis) [inline website/content/en/docs/concepts/disruption.md:290]: Does it make sense to create a sub-bulleted list here?

Comment (jonathan-innis) [inline website/content/en/docs/concepts/disruption.md:290]:
```suggestion
  Karpenter will not be able to complete termination of the node until one of the following conditions is met: the `do-not-disrupt` pods are removed, the pods enter [terminating](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-phase) or [terminal](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-phase) states, or the owning NodeClaim's [TerminationGracePeriod]({{<ref "#terminationgraceperiod" >}}) has been met.
```

File: website/content/en/docs/concepts/disruption.md [hunk 6]
```diff
@@ -274,8 +282,14 @@ Duration and Schedule must be defined together. When omitted, the budget is alwa
 
 ### Pod-Level Controls
 
-You can block Karpenter from voluntarily choosing to disrupt certain pods by setting the `karpenter.sh/do-not-disrupt: "true"` annotation on the pod. This is useful for pods that you want to run from start to finish without disruption. By opting pods out of this disruption, you are telling Karpenter that it should not voluntarily remove a node containing this pod.
+You can block Karpenter from voluntarily choosing to disrupt certain pods by setting the `karpenter.sh/do-not-disrupt: "true"` annotation on the pod.
```
Comment (jonathan-innis) [inline website/content/en/docs/concepts/disruption.md:285]:
```suggestion
You can block Karpenter from voluntarily disrupting pods by setting the `karpenter.sh/do-not-disrupt: "true"` annotation on the pod.
```

File: website/content/en/docs/concepts/disruption.md [hunk 7]
```diff
@@ -200,17 +200,25 @@ To enable interruption handling, configure the `--interruption-queue` CLI argume
 
 ## Controls
 
-### TerminationGracePeriod 
+### TerminationGracePeriod
 
-You can set a NodePool's `terminationGracePeriod` through the `spec.template.spec.terminationGracePeriod` field. This field defines  the duration of time that a node can be draining before it's forcibly deleted. A node begins draining when it's deleted. Pods will be deleted preemptively based on its TerminationGracePeriodSeconds before this terminationGracePeriod ends to give as much time to cleanup as possible. Note that if your pod's terminationGracePeriodSeconds is larger than this terminationGracePeriod, Karpenter may forcibly delete the pod before it has its full terminationGracePeriod to cleanup. 
+You can set a NodePool's `terminationGracePeriod` through the `spec.template.spec.terminationGracePeriod` field.
+This field defines  the duration of time that a node can be draining before it's forcibly deleted.
```
Comment (jonathan-innis) [inline website/content/en/docs/concepts/disruption.md:206]:
```suggestion
This field defines the duration of time that a node can be draining before it's forcibly deleted.
```

File: website/content/en/docs/concepts/disruption.md [hunk 8]
```diff
@@ -200,17 +200,25 @@ To enable interruption handling, configure the `--interruption-queue` CLI argume
...
+A node begins draining when it's deleted.
+Pods will be deleted preemptively based on its TerminationGracePeriodSeconds before this terminationGracePeriod ends to give as much time to cleanup as possible.
+Note that if your pod's terminationGracePeriodSeconds is larger than this terminationGracePeriod, Karpenter may forcibly delete the pod before it has its full terminationGracePeriod to cleanup.
 
-This is especially useful in combination with `nodepool.spec.template.spec.expireAfter` to define an absolute maximum on the lifetime of a node, where a node is deleted at `expireAfter` and finishes draining within the `terminationGracePeriod` thereafter. Pods blocking eviction like PDBs and do-not-disrupt will block full draining until the `terminationGracePeriod` is reached. 
+This is especially useful in combination with `nodepool.spec.template.spec.expireAfter` to define an absolute maximum on the lifetime of a node, where a node is deleted at `expireAfter` and finishes draining within the `terminationGracePeriod` thereafter.
+Pods blocking eviction like PDBs and do-not-disrupt will block full draining until the `terminationGracePeriod` is reached.
```
Comment (jonathan-innis) [inline website/content/en/docs/concepts/disruption.md:212]:
```suggestion
Pods blocking eviction like PDBs and `do-not-disrupt` will block full draining until the `terminationGracePeriod` is reached.
```

File: website/content/en/docs/concepts/disruption.md [hunk 9]
```diff
@@ -200,17 +200,25 @@ To enable interruption handling, configure the `--interruption-queue` CLI argume
...
 
-For instance, a NodeClaim with `terminationGracePeriod` set to `1h` and an `expireAfter` set to `23h` will begin draining after it's lived for `23h`. Let's say a `do-not-disrupt` pod has `TerminationGracePeriodSeconds` set to `300` seconds. If the node hasn't been fully drained after `55m`, Karpenter will delete the pod to allow it's full `terminationGracePeriodSeconds` to cleanup. If no pods are blocking draining, Karpenter will cleanup the node as soon as the node is fully drained, rather than waiting for the NodeClaim's `terminationGracePeriod` to finish.
+For instance, a NodeClaim with `terminationGracePeriod` set to `1h` and an `expireAfter` set to `23h` will begin draining after it's lived for `23h`.
```
Comment (jonathan-innis) [inline website/content/en/docs/concepts/disruption.md:214]:
```suggestion
For instance, a NodeClaim with `terminationGracePeriod` set to `1h` and an `expireAfter` set to `23h` will begin draining `23h` after its creation. The NodeClaim will then be allowed to drain for up to `1h` before its forcefully terminated from the cluster.
```

File: website/content/en/docs/concepts/disruption.md [hunk 10]
```diff
@@ -200,17 +200,25 @@ To enable interruption handling, configure the `--interruption-queue` CLI argume
...
+Let's say a `do-not-disrupt` pod has `TerminationGracePeriodSeconds` set to `300` seconds.
```
Comment (jonathan-innis) [inline website/content/en/docs/concepts/disruption.md:215]:
```suggestion
Let's say a `do-not-disrupt` pod has `TerminationGracePeriodSeconds` set to `300` seconds (`5m`).
```

File: website/content/en/docs/concepts/disruption.md (hunk 4 above)
Comment (jmdeal) [reply website/content/en/docs/concepts/disruption.md]: I didn't take this exact suggestion, but I think the spirit of it is there now. Let me know what you think.

File: website/content/en/docs/concepts/disruption.md [hunk 11]
```diff
@@ -292,18 +315,14 @@ spec:
...
 {{% /alert %}}
 
-Examples of voluntary node removal that will be prevented by this annotation include:
-- [Consolidation]({{<ref "#consolidation" >}})
-- [Drift]({{<ref "#drift" >}})
-- Expiration
-
 {{% alert title="Note" color="primary" %}}
-Voluntary node removal does not include [Interruption]({{<ref "#interruption" >}}) or manual deletion initiated through `kubectl delete node`. Both of these are considered involuntary events, since node removal cannot be delayed.
+The `do-not-disrupt` annotation does **not** exclude nodes from involuntary disruption methods, i.e. [Expiration]({{<ref "#expiration" >}}), [Interruption]({{<ref "#interruption" >}}), and manual deletion (e.g. `kubectl delete node ...`).
```
Comment (njtran) [inline website/content/en/docs/concepts/disruption.md:319]: To users who don't understand that involuntary disruption still goes through the termination controller might get confused on the language here. 

```suggestion
The `do-not-disrupt` annotation does **not** prevent nodes from being involuntarily disrupted, i.e. [Expiration]({{<ref "#expiration" >}}), [Interruption]({{<ref "#interruption" >}}), and manual deletion (e.g. `kubectl delete node ...`), but Karpenter will still respect `do-not-disrupt` pods when draining the node as previously mentioned.
```

File: website/content/en/docs/concepts/disruption.md [hunk 12]
```diff
@@ -292,18 +315,14 @@ spec:
...
 {{% /alert %}}
 
 ### Node-Level Controls
 
-You can block Karpenter from voluntarily choosing to disrupt certain nodes by setting the `karpenter.sh/do-not-disrupt: "true"` annotation on the node. This will prevent disruption actions on the node.
+You can block Karpenter from voluntarily choosing to disrupt certain nodes by setting the `karpenter.sh/do-not-disrupt: "true"` annotation on the node.
+This will prevent voluntary disruption actions against the node.
```
Comment (njtran) [inline website/content/en/docs/concepts/disruption.md:325]: Really only consolidation, not eventual. Worth thinking about how to make this distinction clearly and concisely.

File: website/content/en/docs/upgrading/upgrade-guide.md [hunk 13]
```diff
@@ -51,9 +51,13 @@ Below is the full changelog for v1, copied from the [v1 Migration Upgrade Proced
...
+  * Pods with the `karpenter.sh/do-not-disrupt` annotation now block node termination. Termination of a node with these pods will be blocked until those pods are removed, enter a terminating or terminal state, or the NodeClaims's TerminationGracePeriod has expired.
   * Karpenter's generated NodeConfig now takes precedence when generating UserData with the AL2023 `amiFamily`. If you're setting any values managed by Karpenter in your AL2023 UserData, configure these through Karpenter natively (e.g. kubelet configuration fields).
   * Karpenter now adds a `karpenter.sh/unregistered:NoExecute` taint to nodes in injected UserData when using alias in AMISelectorTerms or non-Custom AMIFamily. When using `amiFamily: Custom`, users will need to add this taint into their UserData, where Karpenter will automatically remove it when provisioning nodes.
+  * Discovered standard AL2023 AMIs will no longer be considered compatible with GPU / accelerator workloads. If you're using an AL2023 EC2NodeClass (without AMISelectorTerms) for these workloads, you will need to select your AMI via AMISelectorTerms (non-alias).
```
Comment (njtran) [inline website/content/en/docs/upgrading/upgrade-guide.md:57]: It is now?

**Pull request closed without merging**